git/cmd.py

```diff
@@ -531,6 +533,12 @@ def execute(self, command,
 
         :param with_stdout: If True, default True, we open stdout on the created process
 
+        :param timeout:
+            To specify a timeout in seconds for the git command, after which the process
+            should be killed. This will have no effect if as_process is set to True. It is
+            set to None by default and will let the process run until the timeout is
+            explicitly specified.
```

Comment:
It might be good to make clear that `SIGKILL` can have negative side-effects on a repository. For example, stale locks in case of `git gc` could render the repository incapable of accepting changes until the lock is manually removed.